backend/battling/forms.py

```diff
@@ -54,6 +54,10 @@ def clean(self):
 
     def save(self, commit=True):
         data = self.clean()
+        pokemon = PokemonTeam.objects.filter(team=self.instance)
+
+        if pokemon:
+            PokemonTeam.objects.filter(team=self.instance).delete()
```

Comment:
You can use `self.instance.pokemons.clear()`
https://docs.djangoproject.com/en/3.2/topics/db/examples/many_to_many/